.github/workflows/dynamic-ci.yml

```diff
@@ -0,0 +1,104 @@
+name: Dynamic CI
+
+on:
+  push:
+    branches:
+      - main
+      - release/[0-9]+.[0-9]+
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  merge_group:
+    types: [checks_requested]
+
+jobs:
+  filter:
+    name: Filter changed files
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      docs: ${{ steps.filter.outputs.docs }}
+      code: ${{ steps.filter.outputs.code }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Detect changed files
+        id: filter
+        uses: dorny/paths-filter@v3
+        with:
+          # 'base' and 'ref' are ignored for pull requests
+          # For merge group events, compare against the target branch (main)
+          base: ${{ github.event.merge_group.base_ref }}
+          # For merge group events, use the merge group head ref
+          ref: ${{ github.event.merge_group.head_ref }}
+          filters: |
+            docs:
+              - '.github/workflows/automatic-doc-checks.yml'
+              - 'docs/**'
```

Comment:
Should we also include the markdown files here?

Reply:
The markdown files (`README.md`, `BUILD.linux.md`, etc) don't get processed/linted with `.github/workflows/automatic-doc-checks.yml`.